### ISSUE TITLE
UI actions on window needing mouse would not work in area not falling under the desktop

### ISSUE THREAD

**Issue by Chandru1210** [TestStack/White#403]
Hi,

i'm new for white concept.i have developed the console application for UI Automation using white it is running perfectly when the system is in active if i lock the system getting error as "UI actions on window needing mouse would not work in area not falling under the desktop" Please help me to resolve the issue. 

Thanks in advance,
Chandru.

**Comment by dpisanu** [comment]
White, like all UI Automation Tools / Frameworks, require an unlocked Desktop and the Screensaver to be inactive.
Unlock your desktop. Simple as that.

**Comment by Chandru1210** [comment]
Could you please share the coding to keep unlock desktop. i need to perform this actions in my non business hrs.

**Comment by dpisanu** [comment]
There is no coding required here.
It's a simple Windows Setting.

Set the Screensaver to None.
In your Power Options do not enable Suspend Mode and deactivate Lock on Idle.

**Comment by Chandru1210** [comment]
Thank you. Please let me know is there any possible to perform the UI automation tool even system get locked ?.

**Comment by dpisanu** [comment]
no, as already stated, it is not possible.
UI Automation is based on programatically generated User Input on a rendered UI.
User Input is done via Keyboard and Mouse and direct interaction via UIA.
For this to work, the UI Objects need to be rendered, on screen and visible.

**Comment by Chandru1210** [comment]
Thanks for your help Daniel.

**Comment by dpisanu** [comment]
can this issue be closed?

**Comment by ypopova96** [comment]
How would that work on VMs then?

**Comment by Roemer** [comment]
Two possibilities:
1. Make sure the VMs resolution is good enough to show all elements and that the VM has an open desktop session.
2. Instead of using the mouse and click, you could use UIA patterns (like Invoke on patterns) which do not need a desktop session. [FlaUI](https://github.com/Roemer/FlaUI) supports all those patterns directly.

**Comment by ypopova96** [comment]
That's very good to know, thank you. 

Would you know if there is another way to trigger the button event other than Click() (still using TestStack White)?

**Comment by Roemer** [comment]
I think White also has support for some patterns, for Invoke see: https://github.com/TestStack/White/blob/7de2f2e4666a9443b658426dd210736bdaddf127/src/TestStack.White/UIItems/UIItem.cs#L318

**Comment by ypopova96** [comment]
Thank you, that seems to have solved my problem. 

My concern was the message I was getting, even when using Invoke(): 

> "UI actions on window needing mouse would not work in area not falling under the desktop"

However, it seems that even the message is displayed, it still goes and invokes the button.

**Comment by Roemer** [comment]
Yes, `Invoke` invokes even if the object to invoke is not visible or even disabled I think.

**Comment by ypopova96** [comment]
Following the same principle, is there a way to select an Item from an ItemList not using the mouse? 

item.Select() seems to need the mouse and item.Invoke() doesn't seem to do anything..

**Comment by Roemer** [comment]
You can get other patterns as well like the invoke pattern in the code linke from above. White was created by always trying to simultate the user behavior as close as possible, which is mostly using the mouse. So you'll have a hard time using White without using the mouse. This is one of the differences to FlaUI, where you choose how you want to interact (eg. by patterns, by mouse, mixed).
